dev, v2.1, v3.0/how-to/upgrade: add DM upgrade advice (#1451)
pingcap/docs: dev/how-to/upgrade/data-migration.md

> - Unless otherwise stated, all the following upgrade examples assume that all the data replication tasks have been stopped before the upgrade and all the replication tasks are restarted manually after DM upgrade is finished.
> - The following shows the upgrade procedure of DM versions in reverse chronological order.

## Upgrade to v1.0.0-rc.1-12-gaa39ff9

### Version information

```bash
Release Version: v1.0.0-rc.1-12-gaa39ff9
Git Commit Hash: aa39ff981dfb3e8c0fa4180127246b253604cc34
Git Branch: dm-master
UTC Build Time: 2019-07-24 02:26:08
Go Version: go version go1.11.2 linux/amd64
```

### Main changes

> Starting from this release, all configurations are rigorously checked, and an unrecognized configuration triggers error. This is to ensure that user always knows exactly what the configuration is.

Reviewer: ```suggestion
Starting from this release, TiDB DM checks all configurations rigorously. Unrecognized configuration triggers an error. This is to ensure that users always know exactly what the configuration is.
```
